hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java

```diff
@@ -1753,7 +1756,28 @@ public ConnectionId(InetSocketAddress address, Class<?> protocol,
     InetSocketAddress getAddress() {
       return address;
     }
-    
+
+    /**
+     * This is used to update the remote address when an address change is detected.  This method
+     * ensures that the {@link #hashCode()} won't change.
+     *
+     * @param address the updated address
+     * @throws IllegalArgumentException if the hostname or port doesn't match
+     * @see Connection#updateAddress()
+     */
+    void setAddress(InetSocketAddress address) {
+      if (!Objects.equals(this.address.getHostName(), address.getHostName())) {
```

Comment:
Should we be permissive of case-insensitive matches?

Or perhaps a subsequent audit of hostname comparisons is in order.

https://www.rfc-editor.org/rfc/rfc4343

Reply:
The hostnames will match since the purpose of this call is only to update the IP address, and It is not intended as a general purpose mutator.  In addition, the protections here are required to ensure that the hashcode does not change, which is based off of the hostname and port.  If the hashcode changes then the connections map will fail to clean-up correctly.